src/hooks/useDelegatesInfo.ts

```diff
@@ -0,0 +1,107 @@
+import useAsyncMemo from 'decentraland-gatsby/dist/hooks/useAsyncMemo'
+import fetch from 'isomorphic-fetch'
+import { useMemo } from 'react'
+import { SNAPSHOT_QUERY_ENDPOINT, SNAPSHOT_SPACE } from '../entities/Snapshot/constants'
+import useVotingPowerBalance from './useVotingPowerBalance'
+
+const QUERY = `
+query ($space: String!, $address: [Bytes!]) {
+  delegatedFrom: delegations(
+    where: {space_in: ["", $space], delegate_in: $address}
+    orderBy: delegate
+    orderDirection: desc
+  ) {
+    delegator
+    delegate
+    space
+    timestamp
+  }
+}
+`
+type DelegationQueryResult = {
+  delegator: string
+  delegate: string
+  space: string
+}
+
+type PickedByResult = {
+  address: string
+  pickedBy: number
+}
+
+export type Delegate = {
+  address: string
+  pickedBy: number
+  totalVP: number
+}
+
+function getTotalVP(addresses: string[]) {
+  return addresses.map(addr => {
+    const [votingPower] = useVotingPowerBalance(addr, SNAPSHOT_SPACE)
+    return { [addr]: votingPower }
+  }).reduce((obj, item) => Object.assign(obj, item), {})
+}
+
+function getPickedBy(addresses: string[]) {
+  const initialValue: PickedByResult[] = []
+  return useAsyncMemo(async () => {
+    if (addresses.length === 0 || SNAPSHOT_SPACE === '') {
+      return initialValue
+    }
+
+    const request = await fetch(
+      SNAPSHOT_QUERY_ENDPOINT,
+      {
+        method: 'post',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({
+          query: QUERY,
+          variables: { address: addresses, space: SNAPSHOT_SPACE },
+        }),
+      }
+    )
+
+    const body = await request.json()
+    const queryResult = body.data.delegatedFrom as DelegationQueryResult[]
+
+    const pickedBy = new Map<string, Set<string>>()
+
+    for (const addr of addresses) {
+
+      const delegations = queryResult.filter(deleg => deleg.delegate === addr)
+      pickedBy.set(addr, new Set())
+
+      if (delegations.length > 0) {
+        for (const deleg of delegations) {
+          pickedBy.get(deleg.delegate)?.add(deleg.delegator)
+        }
+      }
+    }
```

Comment:
Maybe the `picked by` could be calculated by filtering all delegations where the delegate has the address you are querying for, and getting the length?

```typescript
const result = PickedByResult[]
for (const address of addresses) {
    const delegations = queryResult.filter(delegation => delegation.delegate === address)
    result.push({ address, delegations.length })
 }
```

Reply:
In that case you would be double counting someone who delegated in the dcl space and in the global space to the same person